### PR TITLE
Sync .editorconfig with dmd

### PR DESCRIPTION
**Pull request by kubo39**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,9 +1,25 @@
+# EditorConfig file: http://editorconfig.org/
+
 root = true
 
-[*.{c,h,d,di,dd}]
-end_of_line = lf
+# Default settings for all file types:
+# + save files in UTF-8 encoding
+# + use Unix line-endings and a final newline
```
Comment (kinke) [inline .editorconfig:7]: Seems like the Unix line-endings are gone.

File: .editorconfig [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+# + remove whitespace at the end of lines
+[*]
+charset = utf-8
 insert_final_newline = true
+trim_trailing_whitespace = true
+
+# File type specific settings:
+[*.{asm,S,c,cpp,m,h,d,di,dd,ddoc,dt,json,sdl,R,bat,bash,fish,sh,zsh}]
```
Comment (kinke) [inline .editorconfig:15]: The .cpp files should use `indent_size = 2` (one of the clang-format settings).

File: .editorconfig (hunk 1 above)
Comment (kubo39) [reply .editorconfig:7]: DMD removes `end_of_line = lf` in [this PR](https://github.com/dlang/dmd/pull/8656).

Comment (kinke) [reply .editorconfig:7]: Yeah that's probably fine, I meant the comment.

Comment (kubo39) [reply .editorconfig:7]: Ahh, I got!

**Comment by kinke** [comment]
Requesting a review from Johan as he added this file. [Seems like no editor I'm using considers this file.]

**Comment by JohanEngelen** [comment]
Hm, I've never used this file either for an editor. I see in the logs that I added it for `dfmt`. I stopped using dfmt too because the output quality is not sufficient to be used in practice.
So I don't really have an opinion on this file, nor a way to test it.

**Pull request merged**